src/systems/user_commands/UserCommands.cc

```diff
@@ -630,6 +713,125 @@ bool RemoveCommand::Execute()
   return true;
 }
 
+
+//////////////////////////////////////////////////
+LightCommand::LightCommand(msgs::Light *_msg,
+    std::shared_ptr<UserCommandsInterface> &_iface)
+    : UserCommandBase(_msg, _iface)
+{
+}
+
+//////////////////////////////////////////////////
+bool LightCommand::Execute()
+{
+  auto lightMsg = dynamic_cast<const msgs::Light *>(this->msg);
+  if (nullptr == lightMsg)
+  {
+    ignerr << "Internal error, null create message" << std::endl;
+    return false;
+  }
+
+  auto entity = this->iface->ecm->EntityByComponents(
+      components::Name(lightMsg->name()),
+      components::ParentEntity(this->iface->worldEntity));
+
+  auto lightComp = this->iface->ecm->Component<components::Light>(entity);
+  if (nullptr == lightComp)
+  {
+    entity = kNullEntity;
+    // try to find the light inside a link
+    this->iface->ecm->Each<components::Name>(
```

Comment:
this will loop through all the entities in the world since it's checking for entities that have a name. How about doing a search using `EntityByComponents` like above but only passing the filter `lightMsg->name()`? We'll then just verify that the entity's parent is a link or not:


```
    // try to find the light inside a link
    entity = kNullEntity;
    auto lightEnt = this->iface->ecm->EntityByComponents(
        components::Name(lightMsg->name()));
    if (lightEnt != kNullEntity)
    {
      // check if light parent is a link
      auto parentComp = this->iface->ecm->Component<components::ParentEntity>(
          lightEnt); 
      if (parentComp && this->iface->ecm->Component<components::Link>(
          parentComp->Data()))
      {
        lightComp = this->iface->ecm->Component<components::Light>(lightEnt);
        entity = lightEnt;
      }
```